### ISSUE TITLE
How to set up libGDX

### ISSUE THREAD

**Issue by 64BitUniverse** [Catacomb-Snatch/Catacomb-Snatch#8]
I know that the last discussions and comments were three years ago, but I would like to work on this project, but I'm not sure on what to put in libGDX. For Package is it net.catacombsnatch.game? What is the Game class? Do I put the Destination in the same folder where the game files are?

**Comment by spaceemotion** [comment]
Hey! thanks for having interest in working on this title.

It's been a while since I've worked on this title - and java in general - so I can't help you out with the upgrading. This project still runs on java 7 which is like 4 versions behind current the standard it seems. You might be able to re-run the setup program from the libgdx website and copy/paste the old code into the newly created project (just an idea).

The package is `net.catacombsnatch.game` since we still own the domain. Regarding a game class: there isn't really any. The system is composed of a number of scenes/screens. It also uses an entity component system, so if that's something you haven't heard of/ used before, please read up on it beforehand since it's a way to add lots of flexibility without too much overhead.

That being said, internally we've discussed the future of the project a couple times already. In order to bring the game to a lot of devices, moving the engine to Unity would provide the means to publish it on a lot more platforms as well as back it up with an establishes modding community (regardless, we would try to incorporate an easy solution to do so without any hardcore codding knowledge).

Since we're all working our dayjobs we hardly find any time to continue this, but who knows. If you still want to work on the Java version, go forward. If you want to start working on a Unity alternative, that'd also be fine. Your choice.

Good luck!

**Comment by 64BitUniverse** [comment]
Thanks! (RIP Ouya ;-;) Since Gradle wasn't working, I just made a new project and copied it over the old code to there. I'm working on updating code to match new syntax. I'm getting rid of projects that have died since this was created, like Trove or RoboVM. I do agree that Unity is a better engine. For now, I want to get the code to work in Android Studio.

**Comment by 64BitUniverse** [comment]
I made my own fork, and once I don't get any errors, I will change files there.

**Comment by 64BitUniverse** [comment]
Should I keep putting errors I get here, or on my fork?

**Comment by spaceemotion** [comment]
If you have any questions, just comment in this issue for now. No worries.

**Comment by 64BitUniverse** [comment]
Cool. There's a piece of code in net.catacombsnatch.game.animation.Animations that is keeping it from compiling.
 `public Animations(TextureRegion[][] r) {
        animations = new Animation[r.length];
        for (int i = 0; i < r.length; i++) {
            animations[i] = new Animation(0.2f, new Array<>(Art.lordLard[i]), Animation.PlayMode.LOOP);
        }
    }`
and
 'public TextureRegion getKeyFrame(int index) {
        return animations[index].getKeyFrame(stateTime);
    }'
are being problems. The last one doesn't like it that we are making Lord Lard into a texture region, and since I can't find "getKeyFrame" in the API documentation for libGDX, I tried using some of the other methods instead, but they relied on float instead, but then it would be a conversion between an int and a float number, and most compilers don't like that since it can be a "lossy conversion".

Also, in net.catacombsnatch.game.scene.MenuScene, there's a string that was throwing an error, "error: no suitable method found for draw(Object,int,int)
method Batch.draw(Texture,float,float) is not applicable
(argument mismatch; Object cannot be converted to Texture)
method Batch.draw(TextureRegion,float,float) is not applicable
(argument mismatch; Object cannot be converted to TextureRegion)"

The code was
'protected void drawCharacter() {
        getBatch().draw( ani.getKeyFrame(tick), aniX, aniY);
    }'
and Android Studio suggested to do:
'protected void drawCharacter() {
        getBatch().draw(_(Texture)_ ani.getKeyFrame(tick), aniX, aniY);
    }' (Not actually italics in the code).
Is that a good fix? I couldn't find "getBatch" in the API doc also

**Comment by spaceemotion** [comment]
Okay, so all Components are our code, the reason for not finding any documentation about the getKeyFrame() method is that it's ours - nothing to do with native LibGDX.

I don't know what they changed about the animation framework, but the idea was that the component plays a single frame when rendered using the ingame tick (i guess?). The state of the project was just set up so we can get the background + player running. So the only animation working should be the player's (since Lord Lard has been hardcoded).

The `getBatch()` comes from `com.badlogic.gdx.scenes.scene2d.Stage`, maybe look into their documentation instead?

**Comment by 64BitUniverse** [comment]
Yeah, the docs for com.badlogic is where I've been. I was trying to do a general search with ctrl+f

**Comment by 64BitUniverse** [comment]
I've downloaded Unity after purging my drive a bit, and it turns out they only support C# now. Luckily, I've had experience with C++ and know of a place to learn C#, so I'm going to try and rewrite in C. I think it would've needed to be rewritten anyway, since the code relies on libGDX libraries.

**Comment by 64BitUniverse** [comment]
It being in a somewhat different language makes it a bit tougher

**Comment by spaceemotion** [comment]
The thing is: when we would rewrite it in Unity we would use the new ECS system just like we did in LibGDX. If dou have nonprior experience with that I‘d suggest reading up on that first and looking at a few tutorials.

**Comment by 64BitUniverse** [comment]
I noticed that you want to make a HTML version? Microsoft has something called ASP.NET that could do a web app version. Speaking of the internet, is the ces based branch the server branch?

**Comment by spaceemotion** [comment]
Unity can already export for the web. there is no need for asp.net for the web. I also dont know anything about having a server branch in this repo. we had a few server tests, but they are unrelated to this exact state of the project.

**Comment by 64BitUniverse** [comment]
I got some questions about textures:

There are some files in src\main\resources\tiled\tilesets that I have ides on what they are but not entirely sure, and some I have no idea why they are there.
- The tileset-player1.png and tileset-player2.png, are those supposed to be overlaid on the opposition's camp? so, if you are P1, P2's camp would look blue?
- The tileset-overlay.png I feel has no use, except for the texture of the darker rail and the spawners with the mob icon on them. All the other textures are else where in the project. Same with all other files in the folder.


There's a random file called tiles.png, and I'm not sure what that's supposed to be used for. It has the icons for (what seems to me), floor, cancel (or exit), flag, chest, uh... super jump or a lift?, and a bomb.


In src\main\resources\effects there's a file called bullets.png and it has what seems to be a bowl with some of the treasure in it, an empty bowl, a burst effect (I think), and what seems to be a coin.


In src\main\resources\map there's two files I'm not sure about. 
- dark.png looks like it has two foldable texture, like armor stands or mobs in Minecraft, but I'm not sure how it folds and what it's for. 
- mapicons.png has the skull, I get what that's for, but it has a flower, three yellow orbs of different sizes, and a bunch of green and blue squares.

**Comment by spaceemotion** [comment]
Sorry for the silence, currently quite busy with things.

I've checked via the search tool and could not find the files you've listed. Are you referring to the ones in the "original" Mojam fork?

**Comment by 64BitUniverse** [comment]
No problem. Oh yeah, they are. I forgot that I also downloaded the original.

**Comment by spaceemotion** [comment]
1. each player has their own color, yes
2. From what I can see the overlay images are only being used because the project is using tiled. The individual sprites are being used in the game instead.
3. again, tiles.png is just for level editing purposes
4. according to the source "bullets.png" is a misnomer - it contains all the actual sprites still used by the game
5. dark.png is used for the fog of war effect
6. Looking at the code the mapicons have their own index: https://github.com/Maescool/Catacomb-Snatch/search?q=minimapIcon&unscoped_q=minimapIcon
  - Treasurepile
  - Entity
  - Player
  - Spawner

**Comment by 64BitUniverse** [comment]
Thank you. I got some ideas on how to use some of these.

**Comment by Maescool** [comment]
@64BitUniverse if you like, you can contact me by mail (see my profile)
I can give you access to the site repo, and this repo, and the jenkins if you'd like.
I'll keep hosting the things, no problem

**Comment by 64BitUniverse** [comment]
Thanks Maescool. I would like some help with the project, since I'm getting swamped by school, and I burned my self out by working on it too quickly/much 😅 Just couldn't focus on tutorials

**Comment by 64BitUniverse** [comment]
I won't forget about it.

**Comment by 64BitUniverse** [comment]
I got the character moving. Now I need to get Unity to see the mouse.

**Comment by 64BitUniverse** [comment]
Oh, and @Maescool I sent you an email

**Comment by 64BitUniverse** [comment]
I got a question/suggestion about future things. One of the original things was to make a version for iOS right? That's why you used LibGDX? Instead of posting it on the app store, since that's $99 a year for a developer license if I'm not wrong, what it we ask Riley Testut if we could put it on the Alt Store?

**Comment by 64BitUniverse** [comment]
Another question about things I found: How do you guys use bit maps, and how did you store what the maps looked like? Were the walls and floor procedural made every time? I want to recreate the Mojang and DevMap maps so I can work on wall collision and making prebuilt maps.

**Comment by spaceemotion** [comment]
I have/had an iOS developer license that ran out just last (although not because of CS but rather another client of mine). We can get it again no problem.

**Comment by 64BitUniverse** [comment]
Along the lines of the level data, I found these .tmx files along with the bitmaps of the levels, and I tried reading them, but you guys used something called base64 and gzip? How do I decode that?

**Comment by Maescool** [comment]
The original levels are here:
https://github.com/Maescool/Catacomb-Snatch/tree/develop/src/main/resources/levels
the BMP's are the originals.
They used diffrent colors to define things, like dark red for walls, light red for breakable walls.. etc..

**Comment by Maescool** [comment]
tmx files are https://www.mapeditor.org/

**Comment by 64BitUniverse** [comment]
Cool. That's a nice tool.

**Comment by 64BitUniverse** [comment]
Oh wow. There's a thing for Tiled so that I can directly communicate from Tiled to Unity. It's called SuperTiled2Unity

**Comment by 64BitUniverse** [comment]
So, what is seen tile, and how did you randomize the floor and wall tiles?

**Comment by 64BitUniverse** [comment]
I also notice that you didn't have the cornered rails in the Tiled program. Was it a script thing?

**Comment by spaceemotion** [comment]
yes. tiled was making the layout. during import we would do these changes

**Comment by 64BitUniverse** [comment]
@Maescool I sent you an email

**Comment by 64BitUniverse** [comment]
Still working on this. Summer has started, so I'm able to do more. Right now I'm going through some tutorials completely, not just gleaning stuff from them, so I hope to have a better understanding of how Unity's libraries work.

**Comment by 64BitUniverse** [comment]
We now have a font. Don't ask how long it took.

**Comment by 64BitUniverse** [comment]
I also feel dumb for making that font, because it looks like the same characters used in Minecraft, which I didn't realize at the time 😫. The colored characters are what I should've been focusing on instead, but I don't know how to use those as a text with out Photoshoping them into words first then trying to resize them on buttons. I feel like that isn't a very good solution, but until I can come up with a better one, it's what's going to be there

**Comment by 64BitUniverse** [comment]
Does anyone know if there is a debug version of the original game I can access?

**Comment by spaceemotion** [comment]
There are still the java versions up that you can look at I guess

**Comment by 64BitUniverse** [comment]
Ok. I was wondering if there was something that showed hitboxes and the pathing of Astar. It's lucky that most of the 2D AI tutorials for Unity use Astar and the Java version uses Astar.
